src/imagery.config.ts

```diff
@@ -0,0 +1,91 @@
+import * as z from 'zod';
+import { Config, LogType } from '@basemaps/shared';
+import { promises as fs } from 'fs';
+import { ConfigImagery } from '@basemaps/config';
+import { Epsg } from '@basemaps/geo';
+import { production, Updater } from './base.config';
+
+const zBound = z.object({
+  x: z.number(),
+  y: z.number(),
+  width: z.number(),
+  height: z.number(),
+});
+
+const zFile = zBound.extend({
+  name: z.string(),
+});
+
+const zImageConfig = z.object({
+  id: z.string(),
+  name: z.string(),
+  uri: z.string(),
+  projection: z.number(),
+  year: z.number(),
+  resolution: z.number(),
+  bounds: zBound,
+  files: z.array(zFile),
+});
+
+export type ConfigImagerySchema = z.infer<typeof zImageConfig>;
+
+export class ImageryUpdater extends Updater<ConfigImagerySchema, ConfigImagery> {
+  /**
+   * Class to apply an ImageryConfig source to the tile metadata db
+   * @param config a string or ImageryConfig to use
+   */
+  constructor(filename: string, config: unknown, tag: string, isCommit: boolean, logger: LogType) {
+    super(filename, config, tag, isCommit, logger);
+  }
+
+  assertConfig(json: unknown): asserts json is ConfigImagerySchema {
+    zImageConfig.parse(json);
+  }
+
+  async loadOldData(): Promise<ConfigImagery | null> {
+    const oldData = await Config.Imagery.get(this.config.id);
+    return oldData;
+  }
+
+  prepareNewData(oldData: ConfigImagery | null): ConfigImagery {
+    const now = Date.now();
+
+    // Tagging the id.
+    let id = `${this.config.id}@${this.tag}`;
```

Comment:
does this include the `im_` prefix? is there a `Config.Imagery.id` we could use?

Reply:
@blacha , The id in the config file ${this.config.id} is already prefixed.

Reply:
Maybe we should take the `im_` prefix out of the config then, and use the `Config.Imagery.Id` Would be good to be consistent across the types.

Reply:
we removed the im_